pkgs/top-level/all-packages.nix

```diff
@@ -19740,14 +19740,7 @@ with pkgs;
 
   parse-cli-bin = callPackage ../development/tools/parse-cli-bin { };
 
-  patchelf = if with stdenv.buildPlatform; isAarch64 && isMusl then
-    patchelf_0_13
-  else
-    patchelfStable;
-  patchelf_0_13 = callPackage ../development/tools/misc/patchelf/0.13.nix {
-    patchelf = patchelfStable;
-  };
```

Comment:
Attribute is being removed without a throwing alias

Reply:
Is that really mandatory?  Isn't it pretty obvious what's happened if a versioned attribute you were using disappears?  (To the extent it's likely that anybody was explicitly using this version introduced just to be the default on aarch64-musl in the first place…)

Reply:
While the API surface of Nixpkgs isn't explicitly declared somewhere yet, it's pretty much accepted that top-level attributes are effectively part of that. It's also not declared explicitly what stability guarantees Nixpkgs gives about such an API, but I think it's also fairly accepted that attributes should continue existing, that's why we have aliases.

Ideally I'd favor a deprecation message for one release, but in the absence of that we should at least have throwing aliases. Just removing attributes entirely is a bit too much.

And we should assume that any part of the API ends up getting used. Even if we don't hear from them, there likely is some user (or more) out there getting annoyed over attributes randomly disappearing. We can improve those users experience by being more diligent with aliases, I think we should go for that :)

Reply:
Don't bother with this if you don't want to (though I'd merge your PR if you did). But just for the future I think it would be good to have aliases for all removed attributes.

Reply:
> And we should assume that any part of the API ends up getting used. Even if we don't hear from them, there likely is some user (or more) out there getting annoyed over attributes randomly disappearing.

We change little-used APIs all the time.  There's no way it's possible to maintain our ability to iterate if we have to keep every reachable attribute working and unchanging for a two-cycle deprecation period.

Reply:
> if we have to keep every reachable attribute working and unchanging for a two-cycle deprecation period

Yeah, but that's not the case for a throwing alias, which costs effectively nothing